connectors/byoei.py

```diff
@@ -192,6 +197,10 @@ def __init__(
         self.total_docs_created = 0
         self.total_docs_deleted = 0
         self.fetch_error = None
+        self.filtering = filtering
+        self.basic_rule_engine = BasicRuleEngine(
+            parse(filtering.get_active_filter().get("rules", []))
+        )
```

Comment:
Intuitively feels like this syntax will be a bit better:

```suggestion
        self.basic_rule_engine = BasicRuleEngine.parse(filtering.get_active_filter().get("rules", [])))
```

Reply:
The `BasicRuleEngine`'s concern is to execute the rules and not to parse them. Actually `parse` lived inside a `BasicRuleParser` class, but Tarek meant that `parse` should be a top-level function to make it more pythonic as there's no need for an object/class wrapping it. But to me your or Tarek's way is fine tbh.